tests/integration/_experimental/object_detection/test_evaluator_multiclass_test_case_metrics.py

```diff
@@ -0,0 +1,4535 @@
+                Recall=1.0,
+                F1=1.0,
+                AP=0.0,
+            ),
+            ClassMetricsPerTestCase(
+                Class="d",
+                nImages=1,
+                Threshold=0.1,
+                Objects=1,
+                Inferences=1,
+                TP=1,
+                FN=0,
+                FP=0,
+                Precision=1.0,
+                Recall=1.0,
+                F1=1.0,
+                AP=0.0,
+            ),
+        ],
+        Objects=4,
+        Inferences=4,
+        TP=4,
+        FN=0,
+        FP=0,
+        macro_Precision=1.0,
+        macro_Recall=1.0,
+        macro_F1=1.0,
+        mean_AP=0.0,
+    ),
+    (
+        "Threshold: Fixed(0.3), IoU: 0.3, confidence ≥ 0.0",
+        "multiple bboxes in an image, varied iou",
+    ): TestCaseMetricsSingleClass(
+        Objects=4,
+        Inferences=4,
+        TP=3,
+        FN=1,
+        FP=1,
+        Precision=0.75,
+        Recall=0.75,
+        F1=0.75,
+        AP=0.5625,
+    ),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.0",
+        "multiple bboxes in an image, varied iou",
+    ): TestCaseMetricsSingleClass(
+        Objects=4,
+        Inferences=4,
+        TP=2,
+        FN=2,
+        FP=2,
+        Precision=0.5,
+        Recall=0.5,
+        F1=0.5,
+        AP=0.3333333333333333,
+    ),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.3",
+        "multiple bboxes in an image, varied iou",
+    ): TestCaseMetricsSingleClass(
+        Objects=4,
+        Inferences=4,
+        TP=2,
+        FN=2,
+        FP=2,
+        Precision=0.5,
+        Recall=0.5,
+        F1=0.5,
+        AP=0.3333333333333333,
+    ),
+    (
+        "Threshold: F1-Optimal, IoU: 0.5, confidence ≥ 0.1",
+        "multiple bboxes in an image, varied iou",
+    ): TestCaseMetricsSingleClass(
+        Objects=4,
+        Inferences=0,
+        TP=0,
+        FN=4,
+        FP=0,
+        Precision=0.0,
+        Recall=0.0,
+        F1=0.0,
+        AP=0.0,
+    ),
+    (
+        "Threshold: Fixed(0.3) by class, IoU: 0.3, confidence ≥ 0.0",
+        "multiple bboxes in an image, varied iou",
+    ): TestCaseMetrics(
+        PerClass=[
+            ClassMetricsPerTestCase(
+                Class="a",
+                nImages=1,
+                Threshold=0.3,
+                Objects=3,
+                Inferences=3,
+                TP=3,
+                FN=0,
+                FP=0,
+                Precision=1.0,
+                Recall=1.0,
+                F1=1.0,
+                AP=1.0,
+            ),
+            ClassMetricsPerTestCase(
+                Class="b",
+                nImages=1,
+                Threshold=0.3,
+                Objects=1,
+                Inferences=1,
+                TP=0,
+                FN=1,
+                FP=1,
+                Precision=0.0,
+                Recall=0.0,
+                F1=0.0,
+                AP=0.0,
```

Comment:
This still bothers me.  We know that this is incorrect, and having these all over our unit tests seems just wrong to me.  Not plotting a single point PR curve was our decision but I don't think that should affect the AP calculation. A single point precision and recall has a valid AP and I think we should reflect that in our AP calculation. 

At this point, I would rather prefer to plot a single point PR than to have incorrect AP value. What do you think @marko-polo-cheno?

Reply:
We shouldn't plot a PR curve that has a single point because it is noise to the interface. The AP calculation at its core estimates AUC-PR, and it relies on a PR curve.
From internal discussions, we set our AP calculation to include (0, 0) on the PR curve. By this definition, we expect the area to be 0 when there is only a single point, and so the AP value is 0. If we want to see a PR curve with AP = 1 (this [discussion](https://github.com/kolenaIO/kolena/pull/107#discussion_r1253245198)) then we will have to include (recall=0, precision=1) and change our AP calculation util. I don't think that would be a good change.

I know this feels wrong so I suggest that whenever there is no valid PR curve, `AP <-- None` or `AP <-- precision value`. I prefer `None` because a random AP=0.4 is meaningless.

Reply:
Resolved via default average precision method and PR curve fix